Content.Server/Actions/ProjectileSpell.cs

```diff
@@ -0,0 +1,73 @@
+using Content.Server.GameObjects.Components.Projectiles;
+using Content.Server.Utility;
+using Content.Shared.Actions;
+using Content.Shared.GameObjects.Components.Mobs;
+using Content.Shared.Physics;
+using Content.Shared.Utility;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.Actions
+{
+    [UsedImplicitly]
+    public class ProjectileSpell : ITargetPointAction
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        public string CastMessage { get; private set; }
+        public string Projectile { get; private set; }
+
+        public bool Stationary { get; private set; }
+        public float VelocityMult { get; private set; }
+        public float CoolDown { get; private set; }
+        public bool IgnoreCaster { get; private set; }
+
+        public ProjectileSpell()
+        {
+            IoCManager.InjectDependencies(this);
+        }
+
+        public void ExposeData(ObjectSerializer serializer)
+        {
+            serializer.DataField(this, x => x.CastMessage, "castmessage", "Instant action used."); //What player says upon casting the spell
+            serializer.DataField(this, x => x.Projectile, "spellprojectile", null); //What projectile/Entity does the spell create
```

Comment:
add a check below this line to see if the field was set OR give it a proper default value